src/magic_folder/config.py

```diff
@@ -590,28 +679,43 @@ def is_admin(self):
         return not collective_dmd.is_readonly()
 
 
+class ITokenProvider(Interface):
+    """
+    How the configuration retrieves and sets API access tokens.
+    """
+
+    def get():
+        """
+        Retrieve the current token.
+        """
+
+    def rotate():
+        """
+        Change the current token to a new random one.
+        """
+
+
 @attr.s
-class GlobalConfigDatabase(object):
+@implementer(ITokenProvider)
+class FilesystemTokenProvider(object):
     """
-    Low-level access to the global configuration database
+    Keep a token on the filesystem
     """
-    database = attr.ib()  # sqlite3 Connection; needs validator
-    api_token_path = attr.ib(validator=attr.validators.instance_of(FilePath))
+    api_token_path = attr.ib(validator=instance_of(FilePath))
     _api_token = attr.ib(default=None)
 
-    @property
-    def api_token(self):
+    def get(self):
         """
-        Current API token
+        Retrieve the current token.
         """
         if self._api_token is None:
             with self.api_token_path.open('rb') as f:
                 self._api_token = f.read()
         return self._api_token
```

Comment:
Not really related to this branch but just to share - At some point I found it surprising that if I rewrote the token file then the implementation *didn't* respect the change due to the caching here.  I don't think rewriting the token file outside of the control of these classes is a common or even particularly good use case but there's no setter right now and it was a convenient thing to try at one point (until I realized it didn't work).